components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/PublicClientAuthenticator.java

```diff
@@ -80,6 +82,20 @@ public boolean authenticateClient(HttpServletRequest request, Map<String, List>
     public boolean canAuthenticate(HttpServletRequest request, Map<String, List> bodyParams, OAuthClientAuthnContext
             context) {
 
+        List publicClientSupportedGrantTypes = OAuthServerConfiguration.getInstance().
+                getPublicClientSupportedGrantTypesList();
+        List grantTypes = bodyParams.get(GRANT_TYPE);
+
+        if (publicClientSupportedGrantTypes != null && grantTypes != null) {
+            for (Object grantType : grantTypes) {
+                if (!publicClientSupportedGrantTypes.contains(grantType)) {
+                    log.warn("The request has sent with a Grant type : " + grantType + ", which is not allowed for " +
```

Comment:
```suggestion
                    log.warn("The request has sent with the grant type: '" + grantType + "' which is not allowed for " +
```

Reply:
Since the grant types from the request is coming as a list IMO it's better to have `a grant type` rather `the grant type`. WDYT?

Reply:
for a valid oauth request there can only be only one grant type right?

Reply:
yes but that check is happening after this authenticator is executed. As for here there can be multiple grant types.